homeassistant/components/arduino.py

```diff
@@ -0,0 +1,183 @@
+"""
+components.arduino
+~~~~~~~~~~~~~~~~~~
+Arduino component that connects to a directly attached Arduino board which
+runs with the Firmata firmware.
+
+Configuration:
+
+To use the Arduino board you will need to add something like the following
+to your config/configuration.yaml
+
+arduino:
+    port: /dev/ttyACM0
+
+If you are using an original Arduino the port will be named ttyACM*. The exact
+number can be determined with 'ls /dev/ttyACM*' or check your 'dmesg'/
+'journalctl -f' output. Keep in mind that Arduino clones are often using a
+different name for the port (e.g. '/dev/ttyUSB*').
+
+A word of caution: The Arduino is not storing states. This means that with
+every initializsation
+
+TODO:
+- Check Firmata version
+- Check if communication with board is ready
+- Close the connection to the board
+- Add capability to use analog and digital pins as sensors
+- Add PWM output feature
+- I2C
+
+"""
+import logging
+
+from homeassistant.helpers import validate_config
+from homeassistant.helpers.entity import ToggleEntity
+from homeassistant.const import (STATE_ON, STATE_OFF, DEVICE_DEFAULT_NAME)
+
+
+DOMAIN = "arduino"
+DEPENDENCIES = []
+BOARD = None
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup(hass, config):
+    """ Setup the Arduino component. """
+
+    # pylint: disable=global-statement
+    if not validate_config(config,
+                           {DOMAIN: ['port']},
+                           _LOGGER):
+        return False
+
+    try:
+        from PyMata.pymata import PyMata
+
+    except ImportError:
+        _LOGGER.error("Error while importing dependency pyMata.")
+
+        return False
+
+    global BOARD
+    BOARD = ArduinoBoard(config[DOMAIN]['port'])
+
+    return True
```

Comment:
Instead of returning `True`, would it be possible to ask PyMata if it managed to connect to the Arduino?